rse-publish.Rmd

```diff
@@ -0,0 +1,544 @@
+but in other cases you might want to save the reader (and your future self)
+the pain of software installation by making use of a tool
+that can automatically install a specified software environment.
+The most prominent such tool in data science at the moment is [conda](https://docs.conda.io).
+A conda environment can be exported,
+
+```shell
+$ conda env export -n myenv -f myenv.yml
+```
+
+and made available so that readers can use it to install the same environment on their
+own computer:
+
+```
+$ conda env create -f myenv.yml
+```
+
+> **Conda environments**
+>
+> The Python for Atmosphere and Ocean Scientists lesson materials
+> maintained by Data Carpentry have a section devoted to
+> [Software Installation using Conda](https://carpentrieslab.github.io/python-aos-lesson/01-conda/index.html)
+
+Beyond conda there are more complex tools like [Docker][docker] and Nix,
+which can literally install your entire environment (down to the precise operating system)
+on a different computer.
+There’s lots of debate
+about the potential and suitability of these tools as a solution to reproducible research,
+but it’s fair to say that their complexity puts them out of reach
+for many researchers.
+
+### Analysis code
+
+The next item you’ll need to publish is a copy of the code
+written to execute those software packages.
+Depending on the size or complexity of the code you have written,
+and whether you re-use it in multiple projects,
+you may publish script by script
+or create a zip file or tar file that includes everything.
+For example,
+the Makefile fragment below creates `~/archive/meow-2019-02-21.tgz`:
+
+```make
+ARCHIVE=${HOME}/archive
+PROJECT=meow
+TODAY=$(shell date "+%Y-%m-%d")
+SCRIPTS=./Makefile ./bin/*.py ./bin/*.sh
+
+## archive : create an archive of all the scripts used in this run
+archive :
+        @mkdir -p ${ARCHIVE}
+        @tar zcf ${ARCHIVE}/${PROJECT}-${TODAY}.tgz
+```
+
+### Data processing steps
+
+A code library and software description on their own are not much use to a reader;
+they also need to know how that code was actually executed.
+This means including the configuration files (Chapter \@ref(configure)),
+and/or command-line parameters used to generate each key result.
+
+The way in which this information is collected and archived
+depends on how your workflow is constructed. 
+If all of a program's parameters are in a configuration file (Chapter \@ref(configure)),
+then that file can be archived.
+Alternatively, you might need to have your program print out its configuration parameters
+and then use `grep` or a script to extract information from the logfile (Chapter \@ref(logging)).
+
+If your workflow involves executing a series of command line programs,
+then you can keep a log/record of the command line entries
+required to produce a given result.
+For example, the [cmdline-provenance](https://cmdline-provenance.readthedocs.io/en/latest/)
```

Comment:
How widely used is this?

Reply:
Not widely at all. It's a little library I wrote for documenting my own data processing steps. I'd love to include some more widely used examples here - I just don't know what those are.